### ISSUE TITLE
React to mutations in original <datalist>

### ISSUE THREAD

**Issue by petephm** [LeaVerou/awesomplete#16923]
Hi Lea,

Is there a destroy method for Awesomplete?

**Comment by vlazar** [comment]
No.

**Comment by LeaVerou** [comment]
Perhaps we should add one, it would be pretty short I think.

**Comment by petephm** [comment]
I could really use a destroy method

**Comment by LeaVerou** [comment]
Do you want to describe your use case a bit? It would help.

**Comment by petephm** [comment]
Sorry, so I've got a select input which "filters" the autocomplete results of the awesomplete input.

I'm disabling items in the datalist that don't match the selected filter. However making changes to the initial datalist after awesomplete has been initialised doesn't hide those options. 

There's probably a better way but, I was going to destroy the original awesomplete instance and then reinitialise with the disabled options?

**Comment by LeaVerou** [comment]
Yes, there is :) You could set the `.list` property of the Awesomplete instance. However, a better way to solve this would be for us to add a mutation observer that observes changes to the original list's `childList` and automatically updates the list property. @vlazar any chance you have time for this? I'm swamped :(

**Comment by petephm** [comment]
Ahh!! Yea didn't think of that, I need to sleep haha. Yeah that would be awesome. Thanks for your help :)

**Comment by TxHawks** [comment]
A mutation observer would be a nicer solution, but unfortunately, it is
only support in ie11 and up

**Comment by LeaVerou** [comment]
> A mutation observer would be a nicer solution, but unfortunately, it is only support in ie11 and up

Whoever is supporting IE10 and below is wasting their time. Check the actual usage stats, they are about 0.5% for each version of IE.

**Comment by vlazar** [comment]
> @vlazar any chance you have time for this? I'm swamped :(

Same here right now :( I can take a look at this as soon I have time, but we have much older issues to deal with.

> Whoever is supporting IE10 and below is wasting their time. Check the actual usage stats.

LOL, Lea :) https://github.com/LeaVerou/awesomplete/issues/16821#issuecomment-186091284

Does it mean we can drop IE10 support and related hacks in code?

**Comment by LeaVerou** [comment]
> Does it mean we can drop IE10 support and related hacks in code?

Do we have many of those? I think we can wait a bit for that, but let's not expend more energy supporting obsolete crap, especially for features that not everyone uses like this one. Checking if `self.MutationObserver` is defined so that there are no errors in IE10- is sufficient.

**Comment by vlazar** [comment]
As far as I remember these hacks were much simpler without IE10 support https://github.com/LeaVerou/awesomplete/blob/gh-pages/awesomplete.js#L298-L311

**Comment by LeaVerou** [comment]
Oh, because of `__proto__`, right? Well, we did the work now, so no point rushing to revert it. But going forward, when it's a non-critical feature, let's just aim at not throwing errors in IE10.

**Comment by nunoarruda** [comment]
I just got into a situation where I need a way to un-instantiate Awesomplete. I have a country select list and I use Awesomplete to show post codes suggestions only when the country is Australia. So I need to un-instantiate Awesomplete when the country is not Australia.

**Comment by vedraj** [comment]
Does it update datalist if user type a new word ?

**Comment by LeaVerou** [comment]
@petephm hey, why did you close this?
